applications/GeoMechanicsApplication/tests/cpp_tests/test_backward_euler_Pw_scheme.cpp

```diff
@@ -0,0 +1,121 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Richard Faasse
+//
+
+#include "custom_strategies/schemes/backward_euler_quasistatic_Pw_scheme.hpp"
+#include "spaces/ublas_space.h"
+#include "testing/testing.h"
+
+using namespace Kratos;
+using SparseSpaceType = UblasSpace<double, CompressedMatrix, Vector>;
+using LocalSpaceType = UblasSpace<double, Matrix, Vector>;
+
+namespace Kratos::Testing {
+
+KRATOS_TEST_CASE_IN_SUITE(BackwardEulerPwScheme_UpdatesVariablesDerivatives_WhenPredictIsCalled,
+                          KratosGeoMechanicsFastSuite)
+{
+    BackwardEulerQuasistaticPwScheme<SparseSpaceType, LocalSpaceType> scheme;
+    Model model;
+    auto& model_part = model.CreateModelPart("dummy", 2);
+
+    model_part.AddNodalSolutionStepVariable(WATER_PRESSURE);
+    model_part.AddNodalSolutionStepVariable(DT_WATER_PRESSURE);
+
+    constexpr double current_temperature = 1.0;
+    constexpr double previous_temperature = 0.0;
+    constexpr double delta_time = 4.0;
+
+    model_part.GetProcessInfo()[DELTA_TIME] = delta_time;
+    auto p_node = model_part.CreateNewNode(0, 0.0, 0.0, 0.0);
+    p_node->FastGetSolutionStepValue(WATER_PRESSURE) = current_temperature;
```

Comment:
equal, but even clearer to me when comparing to the next line:
p_node->FastGetSolutionStepValue(WATER_PRESSURE, 0) = current_temperature;